inventory-management/src/endpoints/handlers.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * This file should be extended by all the request handlers implemented.
+ * It is the base class that contains the request handling logic for
+ * GET POST, PUT, PATCH, DELETE requests for the resources
+ */
+
+'use strict';
+
+const logger = require('../utils/logger');
+const Model = require('../models/inventory-management/index').Model;
+
+/**
+ * The handler functions for all endpoints defined for the inventory management service
+ */
+class Handlers {
+    constructor(model) {
+        this.model = model || (new Model());
+        this.logger = logger;
+    }
+
+    /**
+    * Check if properties are present in an object, returns an object with a boolean value.
+    * When a property is missing, it also returns the missing property.
+    * @static
+    * @param {array} proplist - an array of properties that you want to check
+    * @param {object} obj - the object to check
+    */
+    static propsPresent(proplist, obj) {
+        for (let prop of proplist) {
+            if (!(prop in obj)) {
+                return { valid: false, missing: prop };
+            }
+        }
+        return { valid: true };
+    }
+
+    /**
+    * Add a product to the inventory
+    * @async
+    * @param {Hapi.request} req - the request object
+    * @param {object} rep - the response toolkit (Hapi.h)
+    */
+    async addProduct(req, rep) {
+        // Some initial error checking
+        if (!req.payload) {
+            return rep.response("Body cannot be empty.").code(400);
+        }
+
+        this.logger.debug(`Handler: Adding product ${JSON.stringify(req.payload)} to inventory`);
+
+        try {
+            // Add the product to the inventory
+            const res = await this.model.addProduct(req.payload);
+            this.logger.debug(`\tResult: ${JSON.stringify(res)}`);
+            return rep.response({ data: res }).code(201);
+        }
+
+        // Catch any database errors (e.g. product not found)
+        catch (err) {
+            this.logger.error(JSON.stringify(err));
+            throw err;
+        }
+    }
+
+    /**
+    * Updates a product with provided id in the inventory
+    * @async
+    * @param {Hapi.request} req - the request object
+    * @param {object} rep - the response toolkit (Hapi.h)
+    */
+    async updateProduct(req, rep) {
```

Comment:
Make sure you do error checking on all of the handlers, and take care of all possible cases. You want to do as much error handling as possible up front to avoid unnecessary DB queries. E.g. what does the payload need to contain in order for the database to be able to update a product?